File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/config/IGBDOptions.java
package ru.kmorozov.gbd.core.config;

import ru.kmorozov.gbd.core.config.options.CtxOptions;

/**
 * Created by km on 06.12.2015.
 */
public interface IGBDOptions {

    String getBookId();

    String getOutputDir();

    String getProxyListFile();

    int getImageWidth();

    boolean reloadImages();

    boolean secureMode();

    String pdfOptions();

    CtxOptions ctxOptions();
}

File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/config/IGBDOptions.kt
package ru.kmorozov.gbd.core.config

import ru.kmorozov.gbd.core.config.options.AuthOptions
import ru.kmorozov.gbd.core.config.options.CtxOptions

/**
 * Created by km on 06.12.2015.
 */
interface IGBDOptions {

    val bookId: String

    val storage: IStorage

    val proxyListFile: String

    val imageWidth: Int

    val debugEnabled: Boolean
        get() = false

    val reloadImages: Boolean

    val scanEnabled: Boolean
        get() = false

    val secureMode: Boolean

    fun pdfOptions(): String

    fun ctxOptions(): CtxOptions

    fun authOptions(): AuthOptions? {
        TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
    }

    fun serverMode(): Boolean {
        return false
    }
}
